src/init.cpp

```diff
@@ -1750,8 +1750,10 @@ bool AppInitMain(NodeContext& node)
 
     fs::path est_path = GetDataDir() / FEE_ESTIMATES_FILENAME;
     CAutoFile est_filein(fsbridge::fopen(est_path, "rb"), SER_DISK, CLIENT_VERSION);
-    // Allowed to fail as this file IS missing on first startup.
-    if (!est_filein.IsNull())
+    // Allowed to fail as this file IS missing on first startup. Also, don't initialize
+    // fee estimation with old data if we don't relay transactions, as they would never be
+    // updated.
+    if (!est_filein.IsNull() && g_relay_txes)
```

Comment:
Why read the file then? You could early check like:
```cpp
// Don't initialize fee estimation if we don't relay transactions.
if (!g_relay_txes) {
    fs::path est_path = GetDataDir() / FEE_ESTIMATES_FILENAME;
    CAutoFile est_filein(fsbridge::fopen(est_path, "rb"), SER_DISK, CLIENT_VERSION);
    if (!est_filein.IsNull()) ::feeEstimator.Read(est_filein);
    fFeeEstimatesInitialized = true;
}
```
This also avoids writing to estimates file in `Shutdown`.

Reply:
I think
```cpp
fFeeEstimatesInitialized = true;
```
should be set anyway ?

What do you think about this conceptually ?

Reply:
I think `fFeeEstimatesInitialized` exists to handle early shutdowns. But if you don't read estimates then writing will clear it right?

Reply:
Hmm right